packages/core/src/debugr/debugr.ts

```diff
@@ -0,0 +1,143 @@
+import EventEmitter from 'events';
+import { EventDispatcher, Events } from '../events';
+import {
+  FormatterPlugin,
+  isFormatterPlugin,
+  Plugin,
+  PluginId,
+  PluginManager,
+  Plugins,
+} from '../plugins';
+import { Logger, LogHandler, TContextBase } from '../logger';
+
+export class Debugr<
+  TContext extends TContextBase = { processId: string },
+  TGlobalContext extends Record<string, any> = {},
+> {
+  private readonly eventDispatcher: EventDispatcher;
+
+  public readonly pluginManager: PluginManager<Partial<TContext>, TGlobalContext>;
+
+  private readonly loggerInternal: Logger<Partial<TContext>, TGlobalContext>;
+
+  private constructor(
```

Comment:
btw: constructory bych určitě nedělal `private` - jednak to znamená, že třídu nemůžeš podědit, a jednak to vynucuje používání statickejch `create()` metod, který maj bejt jen convenience pro případ, že si nechceš věci zadrátovat sám třeba v nějakým svým DI, ale určitě nemaj bejt vynucený

Reply:
oka